v03_pipeline/lib/tasks/reference_data/update_variant_annotations_table_with_updated_reference_dataset_test.py

```diff
@@ -34,6 +35,560 @@
+                        homozygote_count=hl.tint32,
+                        pop=hl.tstr,
+                    ),
+                    faf=hl.tarray(hl.tstruct(faf95=hl.tfloat64)),
+                ),
+                key=['locus', 'alleles'],
+                globals=hl.Struct(
+                    freq_index_dict={'adj': 0, 'XY-adj': 1},
+                    faf_index_dict={'adj': 0},
+                ),
+            ),
+        },
+    },
+    'mpc': {
+        '37': MOCK_MPC_CONFIG,
+        '38': MOCK_MPC_CONFIG,
+    },
+    'primate_ai': {
+        '37': MOCK_PRIMATE_AI_CONFIG,
+        '38': MOCK_PRIMATE_AI_CONFIG,
+    },
+    'splice_ai': {
+        '37': MOCK_SPLICE_AI_CONFIG,
+        '38': MOCK_SPLICE_AI_CONFIG,
+    },
+    'topmed': {
+        '37': MOCK_TOPMED_CONFIG,
+        '38': MOCK_TOPMED_CONFIG,
+    },
+    'hgmd': {
+        '37': {
+            **CONFIG['hgmd']['37'],
+            'source_path': 'gs://seqr-reference-data-private/GRCh37/HGMD/HGMD_Pro_2023.1_hg19.vcf.gz',
+            'custom_import': lambda *_: hl.Table.parallelize(
+                [],
+                hl.tstruct(
+                    locus=hl.tlocus('GRCh38'),
+                    alleles=hl.tarray(hl.tstr),
+                    rsid=hl.tstr,
+                    info=hl.tstruct(
+                        CLASS=hl.tstr,
+                    ),
+                ),
+                key=['locus', 'alleles'],
+                globals=hl.Struct(),
+            ),
+        },
+        '38': {
+            **CONFIG['hgmd']['38'],
+            'source_path': 'gs://seqr-reference-data-private/GRCh38/HGMD/HGMD_Pro_2023.1_hg38.vcf.gz',
+            'custom_import': lambda *_: hl.Table.parallelize(
+                [],
+                hl.tstruct(
+                    locus=hl.tlocus('GRCh38'),
+                    alleles=hl.tarray(hl.tstr),
+                    rsid=hl.tstr,
+                    info=hl.tstruct(
+                        CLASS=hl.tstr,
+                    ),
+                ),
+                key=['locus', 'alleles'],
+                globals=hl.Struct(),
+            ),
+        },
+    },
+    'gnomad_non_coding_constraint': {
+        '38': {
+            'select': {'z_score': 'target'},
+            'source_path': 'gs://seqr-reference-data/GRCh38/gnomad_nc_constraint/gnomad_non-coding_constraint_z_scores.ht',
+            'custom_import': lambda *_: hl.Table.parallelize(
+                [],
+                hl.tstruct(
+                    locus=hl.tlocus('GRCh38'),
+                    alleles=hl.tarray(hl.tstr),
+                    target=hl.tfloat64,
+                ),
+                key=['locus', 'alleles'],
+                globals=hl.Struct(),
+            ),
+        },
+    },
+    'screen': {
+        '38': {
+            **CONFIG['screen']['38'],
+            'source_path': 'gs://seqr-reference-data/GRCh38/ccREs/GRCh38-ccREs.ht',
+            'custom_import': lambda *_: hl.Table.parallelize(
```

Comment:
Are the `custom_import`'s here being used at all?  It feels like the mocking here is a little heavy, so it'd  be great if we could simplify in some way.

Reply:
Ah, I'm following now.  We're actually going through and using these stubbed custom_imports as the table imports to check the versions.

Reply:
Exactly, it's a clunky dict but allows us to execute as much of the actual code as possible, including most of `from_dataset_configs()`. It's better than mocking the return value of `from_dataset_configs`.

Reply:
yep!  I'm on board!  this is great.